### PR TITLE
CEDS-1506 Ensure Each Page knows what "Mode" it is in

### PR DESCRIPTION
**Pull request by ed0906**
This allows us to have custom "Back", "Save & Continue" & "Save & Return" behaviour depending on which "Mode" the page is in e.g. "Amending a Declaration", "Checking Answers", "Editing a Draft"

It is represented as a query param on each Journey page `mode=Amend` 
When the param is not present in the URL it defaults to `NormalMode`

### CODE REVIEW

**Comment by bambuchaAdm** [comment]
What do you think about taking mode only as `JourneyRequest` field not as actionParameter ?

**Comment by ed0906** [comment]
test this please

**Comment by ed0906** [comment]
> What do you think about taking mode only as `JourneyRequest` field not as actionParameter ?

It needs to be possible (and obvious) to link to controllers in a certain mode e.g.

`Redirect(controllers.declaration.routes.SummaryController.displayPage(someMode)`

Therefore it needs to be a parameter on the controller method rather than the request

File: app/views/components/back_link.scala.html [hunk 1]
```diff
@@ -14,8 +14,18 @@
  * limitations under the License.
  *@
 
-@(href: String)(implicit messages: Messages)
+@import models.Mode
+@import models.Mode._
+ 
+@(href: Call, mode: Mode = Mode.Normal)(implicit messages: Messages)
```
Comment (SteveSugden) [inline app/views/components/back_link.scala.html]: 👍

Comment (PatrykRudnicki) [reply app/views/components/back_link.scala.html]: Do we need this default parameter? In practice probably always we're passing this value to the view

Comment (ed0906) [reply app/views/components/back_link.scala.html]: Mode is only required on the journey pages, other pages e.g. Submissions/Drafts all have back buttons too and dont care about the mode

**Comment by ed0906** [comment]
test this please

**Pull request merged**